Update changelog for v10.0.0-beta.17 (#249)
mapbox/mapbox-maps-android: CHANGELOG.md

Mapbox welcomes participation and contributions from everyone.

# 10.0.0-beta.17 - April 12, 2021
## Breaking changes ⚠️
* [Annotation plugin] Rename annotation classes ([#227](https://github.com/mapbox/mapbox-maps-android/pull/227))
* [deps] mapboxMap.queryRenderedFeatures will return a new data class QueriedFeature which will contains additional properties(source (id of the source), sourceLayer (id of the source's layer) ,state (feature's state)) ([#247](https://github.com/mapbox/mapbox-maps-android/pull/247))
* [core] Deprecate MapObserver, MapSnapshotterObserver interfaces and align sync method naming, Map#isMapLoaded, StyleManager#isStyleLoaded
* [core] Remove old map#drag API and the AnimationOptions API

> * [core] Move camera related methods to CameraManager

Reviewer: Does not reflect to Android.

Mapbox welcomes participation and contributions from everyone.

# 10.0.0-beta.17 - April 12, 2021
## Breaking changes ⚠️

> * [Annotation plugin] Rename annotation classes ([#227](https://github.com/mapbox/mapbox-maps-android/pull/227))

Reviewer: Please also add how exactly classes were renamed - we were already asked about it several times.

> * [Annotation plugin] Rename annotation classes(Symbol -> PointAnnotation, Circle -> CircleAnnotation,Line -> PolylineAnnotation,Fill -> PolygonAnnotation) ([#227](https://github.com/mapbox/mapbox-maps-android/pull/227))

Reviewer: Missing renames about options classes. And please align them like that for better readability:
 - Symbol -> PointAnnotation
 -  ...

Author: IMHO it will be too verbose to mention all the details in changelog. For breaking changes, we will add them to the migration guide

Reviewer: Migration guide is for v9 -> v10 migration while those changes affect v10 only.

Reviewer: In general we have section in migration guide about annotations so we must update there as well.
But I would also provide that information here - @pengdev what are your thoughts here?

Reviewer 2: Was it different in v9? if no I think we should state this clearly in the migration guide.
And we could also include a link to the migration guide in this entry when the migration guide is updated?

Author: It's different in v9

> * [deps] mapboxMap.queryRenderedFeatures will return a new data class QueriedFeature which will contains additional properties(source (id of the source), sourceLayer (id of the source's layer) ,state (feature's state)) ([#247](https://github.com/mapbox/mapbox-maps-android/pull/247))

Reviewer: ```suggestion
* [deps] mapboxMap.queryRenderedFeatures will return a new data class QueriedFeature which will contain additional properties: source (id of the source), sourceLayer (id of the source's layer), state (feature's state) ([#247](https://github.com/mapbox/mapbox-maps-android/pull/247))
```

    - Circle -> CircleAnnotation
    - Line -> PolylineAnnotation
    - Fill -> PolygonAnnotation

> * [deps] mapboxMap.queryRenderedFeatures will return a new data class QueriedFeature which will contain additional properties: source (id of the source), sourceLayer (id of the source's layer), state (feature's state) ([#247](https://github.com/mapbox/mapbox-maps-android/pull/247))

Reviewer 2: we can remove the [deps] here

Mapbox welcomes participation and contributions from everyone.

# 10.0.0-beta.17 - April 14, 2021
## Breaking changes ⚠️
* [Annotation plugin] Rename annotation classes, the rules are as follows and applied for Annotations/Options/Managers ([#227](https://github.com/mapbox/mapbox-maps-android/pull/227))
    - Symbol -> PointAnnotation
    - Circle -> CircleAnnotation
    - Line -> PolylineAnnotation
    - Fill -> PolygonAnnotation

> * mapboxMap.queryRenderedFeatures will return a new data class QueriedFeature which will contain additional properties: source (id of the source), sourceLayer (id of the source's layer), state (feature's state) ([#247](https://github.com/mapbox/mapbox-maps-android/pull/247))

Reviewer 3: ```suggestion
* mapboxMap.queryRenderedFeatures will return a new data class QueriedFeature which will contain additional properties ([#247](https://github.com/mapbox/mapbox-maps-android/pull/247)): 
    - source (id of the source)
    - sourceLayer (id of the source's layer)
    - state (feature's state) 
```

* Deprecate MapObserver, MapSnapshotterObserver interfaces and align sync method naming, Map#isMapLoaded, StyleManager#isStyleLoaded
* Remove old map#drag API and the AnimationOptions API
* Don't emit MapIdle event when there is gesture and / or user animation in progress

> * Make overlay plugin experimental ([#233](https://github.com/mapbox/mapbox-maps-android/pull/233))

Reviewer 3: ```suggestion
* Mark overlay plugin experimental ([#233](https://github.com/mapbox/mapbox-maps-android/pull/233))
```

## Features ✨ and improvements 🏁

> * Integrate map engine with NetworkConnectivity API. Map will respect setMapboxStackConnected setting and decide whether http requests can be made.

Reviewer 3: ```suggestion
* Integrate map engine with NetworkConnectivity API. Map will respect `setMapboxStackConnected` setting and decide whether http requests can be made.
```

* Introducing NetworkConnectivity (offline switch).
* Adding NetworkStatus ReachableViaEthernet.
* Added new CameraManager.cameraForCoordinates overload
* Query rendered features for Circle Layer on Terrain
* Reduce binary size of native map shared library

## Bug fixes 🐞
* Fix runtime crash if logo / attribution not enabled ([#240](https://github.com/mapbox/mapbox-maps-android/pull/240))
* Fix distance expression parsing for geometries containing over 20k points
* Fixed holes in the ground for terrain with negative elevation
* Make StyleImageMissing callback a no-op after ImageManager destruction
* Reset unprocessed event queue for destructed renderer
* Fix clipping of fill-extrusions on near plane
* Set 'clusterMaxZoom' to be 'maxZoom-1' when it is not provided
* Fix crash for the case when MapSnapshotter object is destructed from within user provided callback
* Terrain render cache now disabled during property transitions

> * Tile store: handle sparse tilesets and empty tile packs
> * Tile store: handle immutable tile packs

Reviewer 3: ```suggestion
```

Mapbox welcomes participation and contributions from everyone.

# 10.0.0-beta.17 - April 14, 2021
## Breaking changes ⚠️
* [Annotation plugin] Rename annotation classes, the rules are as follows and applied for Annotations/Options/Managers ([#227](https://github.com/mapbox/mapbox-maps-android/pull/227))
    - Symbol -> PointAnnotation
    - Circle -> CircleAnnotation
    - Line -> PolylineAnnotation
    - Fill -> PolygonAnnotation
* mapboxMap.queryRenderedFeatures will return a new data class QueriedFeature which will contain additional properties ([#247](https://github.com/mapbox/mapbox-maps-android/pull/247)): 
    - source (id of the source)
    - sourceLayer (id of the source's layer)
    - state (feature's state) 

> * Deprecate MapObserver, MapSnapshotterObserver interfaces and align sync method naming, Map#isMapLoaded, StyleManager#isStyleLoaded

Reviewer 2: ```suggestion
```

> * Remove old map#drag API and the AnimationOptions API

Reviewer 2: @kiryldz thoughts on this entry? Is this API public-facing?

Reviewer: Yes, it is public, so let's leave it.

* Don't emit MapIdle event when there is gesture and / or user animation in progress
* Make overlay plugin experimental ([#233](https://github.com/mapbox/mapbox-maps-android/pull/233))

## Features ✨ and improvements 🏁

> * Integrate map engine with NetworkConnectivity API. Map will respect setMapboxStackConnected setting and decide whether http requests can be made.
> * Introducing NetworkConnectivity (offline switch).

Reviewer 2: ```suggestion
* Introducing NetworkConnectivity API (offline switch). If setting setMapboxStackConnected(false), the Reachability API will report NotReachable, http requests are being blocked and if running, cancelled.
```

> * Adding NetworkStatus ReachableViaEthernet.

Reviewer 2: ```suggestion
```

* Added new CameraManager.cameraForCoordinates overload

> * Query rendered features for Circle Layer on Terrain

Reviewer 2: ```suggestion
* Added support for query rendered features for Circle Layer on Terrain
```

> * Reduce binary size of native map shared library

Reviewer 2: ```suggestion
* Enable identical code folding and -lto-O3 optimizations to reduce binary size of native map shared library
```

## Bug fixes 🐞

> * Fix runtime crash if logo / attribution not enabled ([#240](https://github.com/mapbox/mapbox-maps-android/pull/240))

Reviewer 2: Fixed a bug that causes map not loading when connected through ethernet.
